pilot/pkg/xds/ep_filters.go

```diff
@@ -46,6 +51,9 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocLbEndpointsAn
 	// remote gateways (if any)
 	filtered := make([]*LocLbEndpointsAndOptions, 0)
 
+	// TODO compute mTLS mode while generating lbEps to get workload-level policy
+	pa := factory.NewPolicyApplier(b.push, b.service.Attributes.Namespace, nil)
```

Comment:
I am thinking it may be feasible to compute this during `buildLocalityLbEndpointsFromShards` to get workload-level coverage. For each unique set of labels on an endpoint, we compute the mTLS mode using a PolicyApplier and store that on the `EndpointBuilder` in a `map[isLbEndpoint_HostIdentifier]mode` since the `EndpointNetworkFilter` must operate on envoy endpoints since it does the weight rebalancing, and I don't want to add filtering logic to `buildLocalityLbEndpointsFromShards` unless we really need to.